### ISSUE TITLE
Unable to override a single config key

### ISSUE THREAD

**Issue by cricket007** [confluentinc/cp-ansible#45]
In the process of making a Vagrantfile, and while I was starting up a single broker, I wanted to limit the log retention and decrease the replicator factors, so I tried this

```ruby
brokers = ["cp-kafka"]
replication_factor = [3, brokers.length].min
ansible.groups = {
  "all:vars" => {"security_mode" => "plaintext"},
  "zookeeper" => ["cp-zk"],
  "broker" => ["cp-kafka"],
  "broker:vars" => {"kafka" => "{'broker': {'config': {"\
    "'log.retention.hours': 48,"\
    "'offsets.topic.replication.factor': #{replication_factor},"\
    "'transaction.state.log.replication.factor': #{replication_factor}"\
    "}}"
},
```

However, upon running, I got the error that stopped at the user-group creation because `kafka.broker` was now overridden with only having a `config` dictionary. It doesn't have the `kafka.broker.group` anymore. 

I have a workaround that is this, but wanted to know if there was an alternative before patching. 

Prefix with `kafka_` and replace `.` with `_`

```ruby
"broker:vars" => {
    "kafka_log_retention_hours" => 48
    ...
```

Update the `defaults.yml`

```yaml
kafka_log_retention_hours: 168
kafka_offsets_topic_replication_factor: 3
kafka_transaction_state_log_replication_factor: 3

kafka:
  broker:
    ...
    config:
      ...
      log.retention.hours: "{{kafka_log_retention_hours}}"
      ...
```

Same error might happen if setting any single property under `kafka => broker` because it seems the group variables aren't merged

**Comment by cotedm** [comment]
@cricket007 does this relate to #46 ? I'm not super familiar with vagrantfiles and ansible but I wonder if you could use host variables? I haven't tried the playbook with group variables yet so there may be some things not merged down properly.

**Comment by cricket007** [comment]
Not directly related, but yes, since I can create a variable number of brokers, I was trying to use a dynamic value for the replication factor. I started only using log retention as a minimal example, but that failed with the config above. 

I have yet to try making a `group_vars` directory for plain Ansible usage, but that was going to be my next attempt, however, it wouldn't be dynamic unless I used Ruby in Vagrant to write that file. 

As for host variables, that worked fine. Vagrant generated this in the inventory

```
cp.kafka.vagrant kafka='{"broker": {"id": 1}}'
```

Which gets parsed via `ast.literal_eval()` in Ansible, so correctly forms a Python dict. 

However, I would prefer [to use the group vars](https://www.vagrantup.com/docs/provisioning/ansible_intro.html#groups-and-group-variables) since that'll make it the same across all the brokers.
